patch-diff-report-tool/src/main/java/com/github/checkstyle/data/DiffReport.java

```diff
@@ -147,10 +147,17 @@ public void getDiffStatistics() {
                 : records.entrySet()) {
             final List<CheckstyleRecord> list = entry.getValue();
             for (CheckstyleRecord rec : list) {
-                statistics.addSeverityRecord(rec.getSeverity(),
-                        CheckstyleReportsParser.DIFF_REPORT_INDEX);
-                statistics.addModuleRecord(rec.getSource(),
-                        CheckstyleReportsParser.DIFF_REPORT_INDEX);
+                if (rec.getIndex() == CheckstyleReportsParser.BASE_REPORT_INDEX) {
```

Comment:
I don't see why we have to remove the `add` records and inline them here.
It seems to me this new logic should be in the `Statistics` class.

Reply:
As it is quite some time I implemented this, I'm not sure anymore.
But I guess because `addSeverityRecord` is also used from other classes.
Before the change, the base and patch report numbers were just added together by calling add record for both of the other records with DIFF_REPORT_INDEX as index.
If you prefer I can of course instead add two new methods `addSeverityDiffRecord` and `addModuleDiffRecord` with the original record index as parameter and then do the `if` within statistics if you prefer.

Reply:
I am ok with either code.